sdk/aqueduct/api_client.py

```diff
@@ -209,46 +224,53 @@ def construct_full_url(self, route_suffix: str, use_https: Optional[bool] = None
 
-    def _test_connection_protocol(self, try_http: bool, try_https: bool) -> bool:
-        """Returns whether the connection uses https. Raises an exception if unable to connect at all.
-
-        First tries https, then falls back to http.
-        """
-        assert try_http or try_https, "Must test at least one of http or https protocols."
-
-        if try_https:
-            try:
-                url = self.construct_full_url(self.LIST_INTEGRATIONS_ROUTE, use_https=True)
-                self._test_url(url)
-                return True
-            except Exception as e:
-                logger().info(
-                    "Testing if connection is HTTPS fails with:\n{}: {}".format(type(e).__name__, e)
-                )
-
-        if try_http:
-            try:
-                url = self.construct_full_url(self.LIST_INTEGRATIONS_ROUTE, use_https=False)
-                self._test_url(url)
-                return False
-            except Exception as e:
-                logger().info(
-                    "Testing if connection is HTTP fails with:\n{}: {}".format(type(e).__name__, e)
-                )
+    def _validate_server_version(self, server_version: str) -> None:
+        """Checks that the SDK and the server versions match."""
+        sdk_version = require("aqueduct-sdk")[0].version
+        if parse_version(server_version) > parse_version(sdk_version):
+            raise ClientValidationError(
+                "The SDK is outdated, it is using version %s, while the server is of version %s. "
+                "Please update your `aqueduct-sdk` package to the appropriate version. If running "
+                "within a Jupyter notebook, remember to restart the kernel."
+                % (sdk_version, server_version)
+            )
+        elif parse_version(server_version) < parse_version(sdk_version):
+            raise ClientValidationError(
+                "The server is outdated, it is using version %s, while the sdk is of version %s. "
+                "Please update your server, or downgrade your SDK so that the versions match."
+                % (server_version, sdk_version)
+            )
 
-        raise ClientValidationError(
-            "Unable to connect to server. Double check that both your API key `%s` and your specified address `%s` are correct. "
-            % (self.api_key, self.aqueduct_address),
-        )
+    def _check_for_server(self, use_https: bool) -> bool:
+        """Check's if the server exists and can be connected to.
 
-    def _test_url(self, url: str) -> None:
-        """Perform a get on the url with default headers, raising an error if anything goes wrong.
+        Raises:
+             ClientValidationError:
+                If the server cannot be found, or if there is a versioning mismatch between server and sdk.
 
-        We don't are about the value of the response, as long as the request succeeds.
         """
+        try:
+            v = self._get_version(use_https=use_https)
```

Comment:
nit: from context should we rename this to `_get_server_version()` together with the variable name?